core/shared/src/main/scala/magnolia.scala

```diff
@@ -134,6 +134,12 @@ object Magnolia {
 
     checkMethod("combine", "case classes", "CaseClass[Typeclass, _]")
 
+    val fullauto = c.prefix.tree.tpe.baseClasses
+      .map(_.asType.toType.decl(TermName("gen")))
```

Comment:
So, right now, the method does not have to be called `gen`. Could we instead check whether the method being invoked (whatever it's called) is `implicit` or not?

Reply:
I think `c.macroApplication` might help.

Reply:
I asked about this in `scala/contributors` because it has stumped me. I'll try that out.

Reply:
nice! That's exactly it.